pkg/api/api.go

```diff
@@ -98,17 +79,15 @@ func summaryJobsByPlatform(report, reportPrev sippyprocessingv1.TestReport, endD
 }
 
 // top failing tests with a bug
-func summaryTopFailingTestsWithBug(topFailingTestsWithBug []sippyprocessingv1.FailingTestResult, resultPrev map[string]sippyprocessingv1.SortedAggregateTestsResult, endDay int) []sippyv1.FailingTestBug {
+func summaryTopFailingTestsWithBug(topFailingTestsWithBug, prevTopFailingTestsWithBug []sippyprocessingv1.FailingTestResult) []sippyv1.FailingTestBug {
 
 	var topFailingTests []sippyv1.FailingTestBug
 
-	allPrev := resultPrev["all"]
-
 	for _, test := range topFailingTestsWithBug {
 		encodedTestName := url.QueryEscape(regexp.QuoteMeta(test.TestName))
 
 		testLink := fmt.Sprintf("%s%s", html.BugSearchUrl, encodedTestName)
-		testPrev := util.GetPrevTest(test.TestName, allPrev.TestResults)
+		testPrev := util.GetTestResult(test.TestName, prevTopFailingTestsWithBug)
```

Comment:
there's no guarantee this test was a top failing test last week.  That's why this looked it up the prv results from the global set of data, not the prevTopFailing list.

Reply:
looks like maybe you figured that out in your subsequent commit

Reply:
hm. well you fixed it for the html but not for the api

Reply:
> hm. well you fixed it for the html but not for the api

tidied up in 94.  We do need to figure out why there's a second API and if it's supposed to be for a single purpose, why it's so broad.

Reply:
the api wasn't created solely for pm score.  the api was created to make the data available to anyone who wanted to access it in a programmatic way for reporting purposes (such as @mfojtik is doing now, as well as the ML team who are doing some analysis)